src/Storages/ObjectStorage/StorageObjectStorageSettings.cpp

```diff
@@ -0,0 +1,104 @@
+#include <Core/BaseSettings.h>
+#include <Core/BaseSettingsFwdMacrosImpl.h>
+#include <Parsers/ASTCreateQuery.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTSetQuery.h>
+#include <Storages/ObjectStorage/StorageObjectStorage.h>
+#include <Storages/ObjectStorage/StorageObjectStorageSettings.h>
+#include <Storages/System/MutableColumnsAndConstraints.h>
+#include <Common/Exception.h>
+
+namespace DB
+{
+
+#define STORAGE_OBJECT_STORAGE_RELATED_SETTINGS(DECLARE, ALIAS) DECLARE(Bool, allow_dynamic_metadata_for_data_lakes, false, "", 0)
```

Comment:
What is the reason of making it a storage setting and not query-level setting?
With query-level setting we can set it to `true` safely because of `compatibility` setting and settings changes history

Reply:
Negotiated that for now we introduce a table-level setting with false default value.